app/models/manageiq/providers/storage_manager.rb

```diff
@@ -12,6 +12,9 @@ class StorageManager < ManageIQ::Providers::BaseManager
     supports_not :cloud_object_store_container_create
     supports_not :ems_storage_new
 
+    has_many :cloud_tenants, :foreign_key => :ems_id, :dependent => :destroy
+    has_many :volume_availability_zones, :foreign_key => :ems_id, :dependent => :destroy
```

Comment:
Since the model name is actually `AvailabilityZone` you have to pass a `:class_name => "AvailabilityZone"` here